### PR TITLE
Add new parsed prop: build context

### PR DESCRIPTION
**Pull request by podviaznikov**
We need octobear to return `build context` property, so that we cab path proper build context into the docker images builder. Previously we just returned path to the Dockerfile.
I also grouped new property and `dockerFilePath` under `build`(so it's back).
So we will need to release major version.
I aslo added one more additional test repo.

This PR also includes fix for git url that has `.git` and branch data.

### CODE REVIEW

**Comment by podviaznikov** [comment]
1__podviaznikov_anton-macpro____runnable_octobear__zsh_

**Review by thejsj — APPROVED** [review]

**Comment by podviaznikov** [comment]
5__podviaznikov_anton-macpro____runnable_octobear__zsh_

File: lib/parser.js [hunk 1]
```diff
@@ -53,15 +53,15 @@ module.exports = class Parser {
...
   /**
-   * Parser for `buildDockerfilePath`
+   * Parser for `dockerFilePath` and `dockerBuildContext`
    *
    * @param {Object}          opts
    * @param {String|Object}   opts.build
    * @param {String|Object}   opts.scmDomain
    * @param {Object}          opts.warnings
-   * @return {String|Null}    buildDockerfilePath - Path for docker build
+   * @return {Object|Null}    object with  `dockerFilePath` and `dockerBuildContext` props
```
Comment (Nathan219) [inline lib/parser.js:62]: Looks like it'll never return a null anymore

File: lib/parser.js [hunk 2]
```diff
@@ -71,24 +71,29 @@ module.exports = class Parser {
...
     // API only takes repo names without .git
-    if (buildPath.substr(buildPath.length - 4) === '.git') {
-      buildPath = buildPath.substr(0, buildPath.length - 4)
+    if (dockerContextPath.substr(dockerContextPath.length - 4) === '.git') {
+      dockerContextPath = dockerContextPath.substr(0, dockerContextPath.length - 4)
     }
     // if path has SCM domain name than we ignore it when creating path or Docker file
-    if (Parser.isScmUrl(scmDomain, buildPath)) {
-      return path.resolve('/', dockerFileName)
+    const dockerFilePath = Parser.isScmUrl(scmDomain, dockerContextPath)
```
Comment (Nathan219) [inline lib/parser.js]: make if, since both ternaries use the same check

**Review by Nathan219 — CHANGES_REQUESTED** [review]

File: lib/parser.js (hunk 1 above)
Comment (podviaznikov) [reply lib/parser.js:62]: see here https://github.com/Runnable/octobear/blob/2cce0b365a9517c6efa1c28f53b517d923dedcc8/lib/parser.js#L68

Comment (podviaznikov) [reply lib/parser.js:62]: see here https://github.com/Runnable/octobear/blob/2cce0b365a9517c6efa1c28f53b517d923dedcc8/lib/parser.js#L68

File: lib/parser.js [hunk 3]
```diff
@@ -71,24 +71,29 @@ module.exports = class Parser {
...
-      buildPath = buildPath.substr(0, buildPath.length - 4)
-    }
+    dockerContextPath = dockerContextPath.replace(/\.git/, '')
     // if path has SCM domain name than we ignore it when creating path or Docker file
-    if (Parser.isScmUrl(scmDomain, buildPath)) {
-      return path.resolve('/', dockerFileName)
+    if (Parser.isScmUrl(scmDomain, dockerContextPath)) {
+      return {
+        dockerFilePath: path.resolve('/', dockerFileName),
+        dockerBuildContext: undefined
```
Comment (Nathan219) [inline lib/parser.js]: lol, why not just not add it?

Comment (podviaznikov) [reply lib/parser.js]: because we don't need it. It not actually build context. It's a remove url

**Review by Nathan219 — APPROVED** [review]

**Comment by podviaznikov** [comment]
1__podviaznikov_anton-macpro____runnable_octobear__zsh_

**Pull request merged**